sites/all/modules/_custom/NC_models/NC_models.install

```diff
@@ -0,0 +1,123 @@
+<?php
+
+function NC_models_schema(){
+	$schema = array();
+
+	$schema['folders'] = array(
+		'description' => 'Table to store details of folders created by users',
+		'fields' => array(
+			'folder_id' => array(
+				'description' => 'Unique folder id which is a combination of folder name and userid of the folder owner or his private folder',
+				'type' => 'varchar',
+				'length' => 150,
+				'not null' => TRUE,
+			),
+			'folder_name' => array(
+				'description' => 'Folder name given by the user',
+				'type' => 'varchar',
+				'length' => 150,
+				'not null' => TRUE,
+			),
+			'owner' => array(
+				'description' => 'User who has created the folder',
+				'type' => 'varchar',
+				'length' => 32,
+				'not null' => TRUE,
+			),
```

Comment:
owner is a foreign key in this table, as in it is a primary key in user table. That should be specified as well.